src/cliffordsynthesis/Tableau.cpp

```diff
@@ -492,4 +492,107 @@ bool Tableau::isIdentityTableau() const {
   }
   return true;
 }
+
+Tableau Tableau::applyMapping(const std::vector<std::vector<bool>>* p) {
+  Tableau mappedTableau = Tableau(nQubits, hasDestabilizers());
+  for (size_t i = 0; i < mappedTableau.getTableauSize(); i++) {
+    for (unsigned char& j : mappedTableau.tableau[i]) {
+      j = 0;
+    }
+  }
```

Comment:
Is the zeroing out necessary? In the next loops, don't you overwrite every entry of the mapped tableau anyway?